### ISSUE TITLE
[BUG] IDEX Incorrect extrude low temp warning and target temp change

### ISSUE THREAD

**Issue by rondlh** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2789]
### Description
For an IDEX system, when extruding from E1 (T1) while E1 is at the required temp, a warning is generated that seems to be referring to the E0 temp. 
When selecting OK in the warning popup, the E1 target temperature is changed to a value that is lower than the value I previously set.

### Steps to reproduce

1. IDEX system, heat T1/E1 to the required temp (200 degrees C)
2. Go to Extruder menu, select E1 and wait until E1 has reached the target temperature (check persistent temp on top)
3. Press Load or Unload
4. A warning message pops up to warn that the temperature is too low, but this message is clearly referring to the E0 temp
5. If I click "OK" then the target temperature of E1 is changed
6. It seems that if I go out of the Extruder menu and go back in that now things seem to be working correctly (no warning when selected extruder temp is ok) 

**Expected behavior**
No low temp warning when selected extruder is at the correct temp.

**Actual behavior**
Incorrect warning, and incorrect target temperature change.

### Hardware Variant
BTT TFT35 V3.0

### TFT Firmware Version & Main Board Firmware details
Latest (2023-05-03)

### Additional Information
I use a low required temp for the extruder of 40 degrees, so I get a reminder, but no need to heat up the nozzle when, for example, loading new filament.

UPDATEE: I think I found the problem, I added 1 line in extrude.c (heatSetCurrentTool(curExtruder_index).

```
    if (extrLength != 0)
    {
      if (curExtruder_index != heatGetCurrentTool()) {
        mustStoreCmd("%s\n", tool_change[curExtruder_index]);
	heatSetCurrentTool(curExtruder_index); // ADDED LINE!!!!!
      }

      switch (warmupNozzle())
```
The added line will immediately inform the TFT to look at the in the TFT selected tool (don't wait for the T0/T1 response, which comes too late).
The same issue is also present in LoadUnload.c and TuneExtruder.c , just search for "switch (warmupNozzle())" only there "curExtruder_index" is called "tool_index"

[3 Patched Files.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/11435000/3.Patched.Files.zip)

**Comment by digant73** [comment]
Fix reported in #2790. Thanks for your work

**Comment by rondlh** [comment]
Tested, it solves the issue, great job again!
